segmentstore/server/src/main/java/io/pravega/segmentstore/server/tables/SegmentKeyCache.java

```diff
@@ -471,7 +463,8 @@ synchronized boolean evict() {
             return false;
         }
 
-        private synchronized byte[] getFromCache() {
+        @GuardedBy("this")
+        private byte[] getFromCache() {
```

Comment:
This is invoked here: https://github.com/pravega/pravega/blob/344180501362fa0f12c1bd98f6e35dd583f26172/segmentstore/server/src/main/java/io/pravega/segmentstore/server/tables/SegmentKeyCache.java#L391
which is not holding the `this` lock. (If that method is supposed to be guarded by `this` (I can't tell if it is supposed to be) it should be private, annotated, and the other use of the lock inside the method should be removed)

Reply:
This was a merge mistake on my end. Fixed now.